packages/odyssey-storybook/src/components/odyssey-mui/Table/Table.stories.tsx

```diff
@@ -0,0 +1,472 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Story } from "@storybook/react";
+import * as React from "react";
+import Button from "@mui/material/Button";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import TableSortLabel from "@mui/material/TableSortLabel";
+import Checkbox from "@mui/material/Checkbox";
+import { MuiThemeDecorator } from "../../../../.storybook/components";
+import TableMdx from "./Table.mdx";
+
+export default {
+  title: `MUI Components/Table`,
+  component: Table,
+  parameters: {
+    docs: {
+      page: TableMdx,
+    },
+  },
+  argTypes: {
+    disabled: {
+      control: "boolean",
+      defaultValue: false,
+    },
+    icon: {
+      control: "text",
+      defaultValue: null,
+    },
+    label: {
+      control: "text",
+      defaultValue: "Asteroids",
+    },
+    wrapped: {
+      control: "boolean",
+      defaultValue: false,
+    },
+  },
+  decorators: [MuiThemeDecorator],
+};
+
+interface Data {
+  name: string;
+  radius: number;
+  type: string;
+  perihelion: string;
+  descriptor: string;
+}
+
+function createData(
+  name: string,
+  radius: number,
+  type: string,
+  perihelion: string,
+  descriptor: string
+): Data {
+  return {
+    name,
+    radius,
+    type,
+    perihelion,
+    descriptor,
+  };
+}
+
+const rows = [
+  createData("Mercury", 2440, "Terrestrial", "--", "Mercurian"),
+  createData("Venus", 6051, "Terrestrial", "--", "Venusian"),
+  createData("Earth", 6378, "Terrestrial", "January 4, 2023", "Terran"),
+  createData("Mars", 3396, "Terrestrial", "June 21, 2022", "Martian"),
+  createData("Jupiter", 71492, "Gas giant", "January 21, 2023", "Jovian"),
+  createData("Saturn", 60268, "Gas giant", "November 29, 2032", "Saturnian"),
+  createData("Uranus", 25559, "Ice giant", "August 17-19, 2050", "Uranian"),
+  createData("Neptune", 24764, "Ice giant", "September 4, 2042", "Neptunian"),
+];
+
+function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
+  if (b[orderBy] < a[orderBy]) {
+    return -1;
+  }
+  if (b[orderBy] > a[orderBy]) {
+    return 1;
+  }
+  return 0;
+}
```

Comment:
I would like to call this `ascendingComparator` because it's confusing to have double-negatives like "not descending" lower down in the code.

Reply:
Do you mind if we keep this while it's purely example code?

I wanted to preserve as much of the MUI code as possible so that devs can cross-check with their docs while we're in the interim state.

Once we wrap this functionality into our own components, I am 💯 on rewriting _all_ of this code to match whatever our prefs are.

Reply:
I also don't like using args where we can use objects instead mainly because objects have named props. Typical comparators only take and A and B arg. In that case, we can use args.

I also don't like using `function` unless we're using `this`.

To get `orderBy` in there, we'd use currying:
```suggestion
const createAscendingComparator = <T>(orderBy: keyof T) => (a: T, b: T) => {
  if (a[orderBy] > b[orderBy]) {
    return 1;
  }
  if (a[orderBy] < b[orderBy]) {
    return -1;
  }
  return 0;
}
```

Use it like this:
```js
[{ id: 3 }, { id: 2 }, { id: 1 }].sort(createAscendingComparator('id'))
```

Example: https://www.typescriptlang.org/play?jsx=0#code/MYewdgzgLgBMBOBTAhlRBBCxFgCYEswBzAYRAFsAHZeVEeGAXhgB4AVAPgAp7dF4AQgE8AXDADWiISABmMNgEomHGF2Ri2AGhgAjDUsYqA3gCgYMfHLUBtXv2EBdGCp234fQUIdLT580igAV3gwGABGAG4zGABfaMtVZDcPR1ZdZPsvH2j-RCCQmABaSOi43PzQgAYouJNQSFgdABtkAAsmGGsjC1wxAGZY7W78XpgAJkGYYdGw2IcAOgh6KC4EFDRMbDxCUgpqWih6LgByEeOFBRM68CWmxHmmkCIuZrbLoA